apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Logging/LoggingTest.cs

```diff
@@ -483,6 +484,78 @@ public async Task Logging_Trace_MultipleSpans()
+                {
+                    // TODO: Uncomment afer fixing https://github.com/googleapis/google-cloud-dotnet/issues/5213
+                    // new object[] { new WebHostBuilder().UseStartup<NoBufferWarningLoggerExternalTraceTestApplication>() },
+                    new object[] { new WebHostBuilder().UseStartup<LoggerNoTracingActivatedTestApplication>() }
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ExternalTraceBuilders))]
+        public async Task Logging_Trace_External_OneEntry(IWebHostBuilder builder)
+        {
+            Timestamp startTime = Timestamp.FromDateTime(DateTime.UtcNow);
+            string testId = IdGenerator.FromGuid();
+
+            string url = $"/Main/Critical/{testId}";
+            using (var server = new TestServer(builder))
+            using (var client = server.CreateClient())
+            {
+                await client.GetAsync(url);
+            }
+
+            _fixture.AddValidator(testId, results =>
+            {
+                // We only have one log entry.
+                LogEntry entry = Assert.Single(results);
+
+                // And the resource name of the trace associated to it contains the external trace id.
+                Assert.Contains(TestEnvironment.GetTestProjectId(), entry.Trace);
+                Assert.Contains("external_trace_id", entry.Trace);
+
+                // The span associated to our entry is the external span.
+                Assert.Equal("external_span_number1", entry.SpanId);
+            });
+        }
+
+        [Theory]
+        [MemberData(nameof(ExternalTraceBuilders))]
+        public async Task Logging_Trace_External_MultipleEntries(IWebHostBuilder builder)
+        {
+            Timestamp startTime = Timestamp.FromDateTime(DateTime.UtcNow);
+            string testId = IdGenerator.FromGuid();
+
+            string url = $"/Main/LogsThreeEntries/{testId}";
+            using (var server = new TestServer(builder))
+            using (var client = server.CreateClient())
+            {
+                await client.GetAsync(url);
+            }
+
+            _fixture.AddValidator(testId, results =>
+            {
+                string projectId = TestEnvironment.GetTestProjectId();
+
+                // We have three log entries.
+                Assert.Equal(3, results.Count);
+
+                // And the resource name of the trace associated to them contains the external trace id.
+                Assert.DoesNotContain(results, entry => !entry.Trace.Contains(projectId));
+                Assert.DoesNotContain(results, entry => !entry.Trace.Contains("external_trace_id"));
+
+                // The span associated to our entry is the external span, and is the same span number
+                // as the log entry.
+                Assert.DoesNotContain(results, entry => entry.SpanId != $"external_span_number{entry.JsonPayload.Fields["message"].StringValue.Last()}");
```

Comment:
The double negative is confusing here, and despite a quick search, I still don't actually know what it means.
Is this effectively an Assert.All?